deployment/terraform/assets/cluster.tf

```diff
@@ -82,6 +82,47 @@ resource "aws_instance" "app_server" {
   }
 }
 
+data "aws_vpc" "selected" {
+  tags = {
+    Name = "Default VPC"
+  }
+}
+
+data "aws_subnets" "loadtest_subnets" {
+  filter {
+    name = "vpc-id"
+    values = [data.aws_vpc.selected.id]
+  }
+
+  tags = {
+    Name = "loadtest*"
+  }
+}
+
+resource "aws_efs_file_system" "efs_shared" {
+  count = var.create_efs ? 1 : 0
+
+  tags = {
+   Name = "Shared-fs-${var.cluster_name}"
+  }
+}
+
+resource "aws_efs_mount_target" "efs_mount" {
+  for_each = (var.create_efs ? toset(data.aws_subnets.loadtest_subnets.ids) : toset({}))
```

Comment:
Why do we need `for_each` here? Can't we use `count = var.create_efs ? 1 : 0` as well?

Reply:
@agarciamontoro We need to create mount target for each availability zone (by using the various subnets).